10.1/root/usr/share/container-scripts/mysql/README.md

```diff
@@ -124,6 +124,42 @@ variable with the full path of the file you wish to use. For example, the defaul
 location is `/etc/my.cnf` but you can change it to `/etc/mysql/my.cnf` by setting
  `MYSQL_DEFAULTS_FILE=/etc/mysql/my.cnf`
 
+Upgrading and datadir checking
+------------------------------
+MySQL and MariaDB use versions that consist of three numbers (e.g. 5.6.23). When two
+versions are different only in the last part, they are compatible, but when upgrading
+from e.g. 5.6 to 5.7 or from 10.0 to 10.1, we need to consider upgrading as described
+in https://dev.mysql.com/doc/refman/5.7/en/upgrading-from-previous-series.html.
+
+Important: Upgrading to a new version is always risky and users are expected to make a full
+back-up of all data before.
+
+A safer solution to upgrade is to dump all data using `mysqldump` or `mysqldbexport` and then
+load the data using `mysql` or `mysqldbimport` into an empty (freshly initialized) database.
+
+Another way of proceeding with the upgrade is starting the new version of the `mysqld` daemon
+and run `mysql_upgrade` right after the start. This so called in-place upgrade is only possible
+if upgrading from the very previous version, so skipping versions is not possible.
+
+This container tries to detect whether the data needs to be upgraded using `mysql_upgrade` and
+we can control it by setting `MYSQL_UPGRADE` variable, which can have the following values:
+
+ * `warn` -- If the data version can be determined and the data come from a different version
+   of the daemon, a warning is printed but the container starts. This is the default value.
```

Comment:
In my opinion running `mysql_upgrade` by default might actually be a bit better (=less evil) in this case, since at least this way the fail would occur at the moment of upgrade and not far off in the future as with the ignored warning (and probably be less/not-at-all destructive?).
Best case would be if there were no :latest tag in the imagestream for DB images but I am not sure if that is possible or even wanted by Openshift.